### PR TITLE
Document temporary indexes

### PR DESCRIPTION
**Pull request by AudriusButkevicius**

### CODE REVIEW

File: specs/bep-v1.rst [hunk 1]
```diff
@@ -844,6 +862,144 @@ XDR
...
     }
 
+DownloadProgress (Type = 8)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The DownloadProgress message is used to notify remote devices about partial
+availability of files. Unlike other messages in the protocol, these are
+incremental, which means they are only sent when progress has been made or the
+state of the file being downloaded has changed. By default, these messages are
+sent every 5 seconds, and only in the cases where progress or state chagnes
```
Comment (phil-davis) [inline specs/bep-v1.rst]: s/chagnes/changes

File: specs/bep-v1.rst [hunk 2]
```diff
@@ -844,6 +862,144 @@ XDR
...
+::
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                           Reserved                          |F|
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+:Bit 31 ("F", Forget): is set to notify that the file that was previously
+    advertised is no longer available (atleast as a temporary file).
```
Comment (phil-davis) [inline specs/bep-v1.rst]: s/atleast/at least/

File: specs/bep-v1.rst [hunk 3]
```diff
@@ -844,6 +862,144 @@ XDR
...
 
+DownloadProgress (Type = 8)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The DownloadProgress message is used to notify remote devices about partial
+availability of files. Unlike other messages in the protocol, these are
+incremental, which means they are only sent when progress has been made or the
+state of the file being downloaded has changed. By default, these messages are
+sent every 5 seconds, and only in the cases where progress or state chagnes
+have been detected. Each DownloadProgress message is addresses to a specific
```
Comment (phil-davis) [inline specs/bep-v1.rst]: s/addresses/addressed/

File: specs/bep-v1.rst [hunk 4]
```diff
@@ -844,6 +862,144 @@ XDR
...
+
+The **Name** field defines the file name from the global index for which this
+update is being sent.
+
+The **Version** structure defines the version of the file for which this update
+is being sent.
+
+**Block Indexes** is a list of positive integers, where each integer represents
+the index of the block in the FileInfo structure Blocks array that has become
+available for downloads.
```
Comment (phil-davis) [inline specs/bep-v1.rst]: s/downloads/download/

File: specs/bep-v1.rst [hunk 5]
```diff
@@ -844,6 +862,140 @@ XDR
         int Code;
     }
 
+DownloadProgress (Type = 8)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The DownloadProgress message is used to notify remote devices about partial
+availability of files. DownloadProgress messages are only sent when progress 
+has been made or the state of the file being downloaded has changed. By default,
+these messages are sent every 5 seconds Each DownloadProgress message is for
```
Comment (calmh) [inline specs/bep-v1.rst]: Missing punctuation after seconds

File: specs/bep-v1.rst [hunk 6]
```diff
@@ -844,6 +862,140 @@ XDR
...
     }
 
+DownloadProgress (Type = 8)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The DownloadProgress message is used to notify remote devices about partial
+availability of files. DownloadProgress messages are only sent when progress 
+has been made or the state of the file being downloaded has changed. By default,
+these messages are sent every 5 seconds Each DownloadProgress message is for 
+a specific folder and MAY contain zero or more FileDownloadProgressUpdate
```
Comment (calmh) [inline specs/bep-v1.rst]: Nothing MAY about it, it has no choice but to contain zero or more FileDownloadProgressUpdates :)

File: specs/bep-v1.rst [hunk 7]
```diff
@@ -844,6 +862,140 @@ XDR
...
+update is being sent.
+
+The **Version** structure defines the version of the file for which this update
+is being sent.
+
+**Block Indexes** is a list of positive integers, where each integer represents
+the index of the block in the FileInfo structure Blocks array that has become
+available for download.
+For example an integer with with value 3 represents that the data defined in the
+third BlockInfo structure of the FileInfo structure of that file is now available.
```
Comment (calmh) [inline specs/bep-v1.rst]: Actually the fourth, assuming the blocks are indexed 0, 1, 2, 3 as usual.

File: specs/bep-v1.rst [hunk 8]
```diff
@@ -844,6 +862,140 @@ XDR
...
+Please note that matching should be done on **Name** AND **Version**.
+Furthermore, each update received is incremental, for example the initial 
+FileDownloadProgressUpdate structure might contain indexes 0, 1, 2, an update 
+5 seconds later might contain indexes 3, 4, 5 which should be appended to the 
+original list, which implies that blocks 0-6 are currently available.
```
Comment (calmh) [inline specs/bep-v1.rst]: 0-5, as those are the blocks we've announced...

File: specs/bep-v1.rst [hunk 9]
```diff
@@ -844,6 +862,140 @@ XDR
...
+
+Blocks are downloaded in random order, therefore block indexes MAY NOT grow in
```
Comment (calmh) [inline specs/bep-v1.rst]: Skip the part about blocks being downloaded in random order as that's implementation dependent, but the conclusion about the indexes not growing in any defined order should be in here as it's true.

File: specs/bep-v1.rst [hunk 10]
```diff
@@ -844,6 +862,140 @@ XDR
...
+an incrementing order.
+
+Value of **Version** changing between update messages implies that the file has
+changed, and that any indexes previously advertised are no longer available.
+The list of available block indexes MUST BE replaced (rather than appended)
```
Comment (calmh) [inline specs/bep-v1.rst]: I think we should keep this, but also add something that if an update for version A has been sent, an implementation MUST send a Forget for version before sending an update for another version B

**Comment by AudriusButkevicius** [comment]
Addressed.

File: specs/bep-v1.rst [hunk 11]
```diff
@@ -844,6 +862,142 @@ XDR
...
+available for download.
+For example an integer with with value 3 represents that the data defined in the
+fourth BlockInfo structure of the FileInfo structure of that file is now available.
+Please note that matching should be done on **Name** AND **Version**.
+Furthermore, each update received is incremental, for example the initial 
+FileDownloadProgressUpdate structure might contain indexes 0, 1, 2, an update 
+5 seconds later might contain indexes 3, 4, 5 which should be appended to the 
+original list, which implies that blocks 0-5 are currently available.
+
+Block indexes MAY NOT grow in any defined order.
```
Comment (calmh) [inline specs/bep-v1.rst]: This forbids the block list to grow in any kind of defined order. :) I think we mean `Block indexes MAY be added in any order. An implementation MUST NOT assume that block indexes are added in any specific order.`

**Comment by AudriusButkevicius** [comment]
FUUU, I've overwritten the changes I've done through the web editor X_x

**Comment by calmh** [comment]
@st-review merge it yo

**Comment by st-review** [comment]
:ok_hand: Merged as de1d3792c1ac8339e33dde13cd1f272eb9c509ee. Thanks, @AudriusButkevicius!

**Pull request closed without merging**